src/transaction/boot_cl.c

```diff
@@ -726,6 +710,142 @@ boot_initialize_client (BOOT_CLIENT_CREDENTIAL * client_credential, BOOT_DB_PATH
+	      return ER_BO_CWD_FAIL;
+	    }
+	}
+    }
+
+  // convert to absolute path
+  char realpath_buf[PATH_MAX];
+  if (realpath (path, realpath_buf) != nullptr)
+    {
+      path = realpath_buf;
+    }
+  else
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_BO_DIRECTORY_DOESNOT_EXIST, 1, path);
+      return ER_BO_DIRECTORY_DOESNOT_EXIST;
+    }
+
+  // remove useless PATH_SEPARATOR
+  char remove_useless_sep_buf[PATH_MAX];
+  boot_remove_useless_path_separator (path, remove_useless_sep_buf);
+  path = remove_useless_sep_buf;
+  
+  // done
+  strcpy (path_out, path);
+  return NO_ERROR;
+}
+
+static int
+boot_initialize_lob_path (const char *path_arg, const char *db_path, char *path_out)
+{
+  // Process the path argument and generate a normalized path containing the external storage prefix.
+  char es_path[PATH_MAX];
+  ES_TYPE es_type = ES_NONE;
+
+  if (path_arg == NULL)
+    {
+      snprintf (es_path, sizeof (es_path), "%s%s%clob", LOB_PATH_DEFAULT_PREFIX, db_path,
+		PATH_SEPARATOR);
+      es_type = ES_DEFAULT_TYPE;
+    }
+  else
+    {
+      es_type = es_get_type (path_arg);
+      switch (es_type)
+	{
+	case ES_NONE:
+	  /* prepend default prefix */
+	  snprintf (es_path, sizeof (es_path), "%s%s", LOB_PATH_DEFAULT_PREFIX, path_arg);
+	  es_type = ES_DEFAULT_TYPE;
+	  break;
+#if !defined (CUBRID_OWFS)
+	case ES_OWFS:
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ES_INVALID_PATH, 1, path_arg);
+	  return ER_ES_INVALID_PATH;
+#endif /* !CUBRID_OWFS */
+	default:
+	  strncpy_bufsize (es_path, path_arg);
+	  break;
+	}
+    }
+
+  assert (es_type != ES_NONE);
+
+  // Get the path without the external storage prefix; the prefix ends with ':'
+  char *lob_dir_path = std::strchr (es_path, ':');
+  if (lob_dir_path == nullptr)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ES_INVALID_PATH, 1, es_path);
+      return ER_ES_INVALID_PATH;
+    }
+  // Move after ':'
+  ++lob_dir_path;
+
+  if (es_get_type (es_path) == ES_POSIX)
+    {
+      // Convert to absolute path and check it exists
+      char absolute_path_buf[PATH_MAX];
+
+#if defined (WINDOWS)
+      struct stat stat_buf;
+      if (realpath (lob_dir_path, absolute_path_buf) != NULL
+	  && (stat (absolute_path_buf, &stat_buf) == 0 && S_ISDIR (stat_buf.st_mode)))
+#else
+      if (realpath (lob_dir_path, absolute_path_buf) != NULL)
+#endif
+	{
+	  // Conversion successful
+	}
+      else
+	{
+	  // Directory does not exist
+	  er_set (ER_NOTIFICATION_SEVERITY, ARG_FILE_LINE, ER_BO_DIRECTORY_DOESNOT_EXIST, 1, lob_dir_path);
+	  if (mkdir (lob_dir_path, 0777) < 0)
+	    {
+	      cub_dirname_r (lob_dir_path, absolute_path_buf, PATH_MAX);
+	      er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ES_GENERAL, 2, "POSIX", absolute_path_buf);
+	      return ER_ES_GENERAL;
+	    }
+	}
+
+      // Remove useless separators and save the value to lob_dir_path (inside es_path)
+      boot_remove_useless_path_separator (absolute_path_buf, lob_dir_path);
```

Comment:
Is this still needed? Doesn't `realpath` remove unneeded separators?

Reply:
I am not sure; I assume it does not, since this is inherited from existing code.